packages/graphql/src/classes/subscription/cdc/cdc-event-parser.ts

```diff
@@ -0,0 +1,123 @@
+        const timestamp = metadata.txStartTime.toStandardDate().getTime();
+
+        if (cdcEvent.eventType === "n") {
+            return this.parseNodeEvent(cdcEvent, timestamp);
+        } else {
+            // Relationship not supported
+            return undefined;
+            // return this.parseRelationshipEvent(cdcEvent, timestamp);
+        }
+    }
+
+    private parseNodeEvent(cdcEvent: CDCNodeEvent, timestamp: number): NodeSubscriptionsEvent | undefined {
+        const typenames = this.getTypenamesFromLabels({
+            labels: cdcEvent.labels,
+            schemaModel: this.schemaModel,
+        });
+        if (!typenames || !typenames[0]) return undefined; // What happens with multiple typenames?
+
+        const typename = typenames[0];
+        switch (cdcEvent.operation) {
+            case "c":
+                return {
+                    event: "create",
+                    typename,
+                    properties: {
+                        old: undefined,
+                        new: serializeProperties(cdcEvent.state.after?.properties) || {},
+                    },
+                    id: cdcEvent.elementId as any,
+                    timestamp,
+                };
+            case "d":
+                return {
+                    event: "delete",
+                    typename,
+                    properties: {
+                        old: serializeProperties(cdcEvent.state.before?.properties) || {},
+                        new: undefined,
+                    },
+                    id: cdcEvent.elementId as any,
+                    timestamp,
+                };
+
+            case "u":
+                return {
+                    event: "update",
+                    typename,
+                    properties: {
+                        old: serializeProperties(cdcEvent.state.before?.properties) || {},
+                        new: serializeProperties(cdcEvent.state.after?.properties) || {},
+                    },
+                    id: cdcEvent.elementId as any,
+                    timestamp,
+                };
+        }
+    }
+
+    private parseRelationshipEvent(
+        _cdcEvent: CDCRelationshipEvent,
+        _timestamp: number
+    ): RelationshipSubscriptionsEvent | undefined {
+        return undefined; // Relationship Events are ignored for now
+        // const type = cdcEvent.type;
+        // const startTypenames = this.getTypenamesFromLabels({
+        //     labels: cdcEvent.start.labels,
+        //     schemaModel: this.schemaModel,
+        // });
+        // if (!startTypenames || !startTypenames[0]) return undefined; // What happens with multiple typenames?
+
+        // // const startTypename = startTypenames[0];
+        // const startEntity = this.schemaModel.getEntitiesByLabels(cdcEvent.start.labels);
+
+        // // TODO: Get relationship
+        // console.log(cdcEvent);
+        // return undefined;
```

Comment:
is this comment needed?

Reply:
mmm, not really, but this is the logic to capture relationship events, of which only the start/end node properties is missing

Reply:
I see